website/www/site/content/en/documentation/runtime/environments.md

```diff
@@ -17,147 +17,255 @@ limitations under the License.
+   You can also provide a custom repo/tag as [additional parameters](#additional-build-parameters).
+
+  ```
+  export BEAM_SDK_VERSION="2.26.0"
+  export IMAGE_NAME="gcr.io/my-gcp-project/beam_python3.7_sdk"
+  export TAG="${BEAM_SDK_VERSION}-custom"
+
+  docker tag apache/beam_python3.7_sdk "${IMAGE_NAME}:${TAG}"
+  docker push "${IMAGE_NAME}:${TAG}"
+  ```
+
+6. After pushing a container image, verify the remote image ID and digest matches the local image ID and digest output from `docker_images --digests`.
+
+#### Additional build parameters{#additional-build-parameters}
+
+The docker Gradle task defines a default image repository and [tag](https://docs.docker.com/engine/reference/commandline/tag/) is the SDK version defined at [gradle.properties](https://github.com/apache/beam/blob/master/gradle.properties). The default repository is the Docker Hub `apache` namespace, and the default tag is the [SDK version](https://github.com/apache/beam/blob/master/gradle.properties) defined at gradle.properties.
+
+You can specify a different repository or tag for built images by providing parameters to the build task. For example:
 
-1. Pull a [prebuilt SDK container image](https://hub.docker.com/search?q=apache%2Fbeam&type=image) for your [target](https://docs.docker.com/docker-hub/repos/#searching-for-repositories) language and version. The following example pulls the latest Python SDK:
 ```
-docker pull apache/beam_python3.7_sdk
+./gradlew :sdks:python:container:py36:docker -Pdocker-repository-root="example-repo" -Pdocker-tag="2.26.0-custom"
 ```
-2. [Write a new Dockerfile](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) that [designates](https://docs.docker.com/engine/reference/builder/#from) the original as its [parent](https://docs.docker.com/glossary/?term=parent%20image).
-3. [Build](#building-container-images) a child image.
 
-### Modifying the original Dockerfile {#modifying-dockerfiles}
+builds the Python 3.6 container and tags it as `example-repo/beam_python3.6_sdk:2.26.0-custom`.
+
+From Beam 2.21.0 and later, a `docker-pull-licenses` flag was introduced to add licenses/notices for third party dependencies to the docker images. For example:
 
-1. Clone the `beam` repository:
 ```
-git clone https://github.com/apache/beam.git
+./gradlew :sdks:java:container:java8:docker -Pdocker-pull-licenses
 ```
-2. Customize the [Dockerfile](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile). If you're adding dependencies from [PyPI](https://pypi.org/), use [`base_image_requirements.txt`](https://github.com/apache/beam/blob/master/sdks/python/container/base_image_requirements.txt) instead.
-3. [Reimage](#building-container-images) the container.
+creates a Java 8 SDK image with appropriate licenses in `/opt/apache/beam/third_party_licenses/`.
 
-### Testing customized images
+By default, no licenses/notices are added to the docker images.
+
+
+## Running pipelines with custom container images
```

Comment:
Can you add an anchor to this so I can link to it in cloud docs please?

Reply:
added #{running-pipelines}